### PR TITLE
Created order controller

### PR DESCRIPTION
**Pull request by GlebPashko**

### CODE REVIEW

File: src/main/java/org/example/bookstore/dto/Empty.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package org.example.bookstore.dto;
+
+public class Empty {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/dto/Empty.java:3]: remove class

File: src/main/java/org/example/bookstore/model/Order.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+@Getter
+@Setter
+@Entity
+@Table(name = "orders")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/Order.java:30]:
```suggestion
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
```

File: src/main/java/org/example/bookstore/model/OrderItem.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+@Getter
+@Setter
+@Entity
+@Table(name = "order_items")
+public class OrderItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/OrderItem.java:23]: add JoinColumn

File: src/main/java/org/example/bookstore/model/OrderItem.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    private Order order;
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/OrderItem.java:25]: add JoinColumn

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId());
+
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Order.Status.PROCESSING);
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(requestDto.getShippingAddress());
+
+        Set<OrderItem> orderItems = orderItemMapper.toOrderItemSet(shoppingCart.getCartItems());
+        orderItems.forEach(o -> o.setOrder(order));
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java:50]: for each is terminal operation `orderItems.forEach(o -> o.setOrder(order));`
Check please that you don't get an exception when you try to use 
`orderItems.forEach(o -> o.setPrice(o.getPrice().multiply(new BigDecimal(o.getQuantity()))));`

Comment (Elena-Bruyako) [reply src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java:50]: Also, rename o

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java (hunk 5 above)
Comment (GlebPashko) [reply src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java]: I checked, works without errors

Comment (GlebPashko) [reply src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java]: for each is terminal operation orderItems.forEach(o -> o.setOrder(order));
Yes, but I'm changing the value in an already created object, so I think it's okay. Correct it if it's wrong, please

File: src/main/java/org/example/bookstore/controller/OrderController.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+@Tag(name = "Order management", description = "Endpoints for managing orders")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+
+    @Operation(summary = "Create a new order")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping
```
Comment (liliia-ponomarenko) [inline src/main/java/org/example/bookstore/controller/OrderController.java:32]: Better use ResponseStatus.CREATED

File: src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+    @Override
+    public OrderItemResponseDto getOrdersItem(Long orderId, Long itemId) {
+        OrderItem orderItem = orderItemRepository.findByOrderIdAndOrderItemId(orderId, itemId);
+        return orderItemMapper.toOrderItemResponseDto(orderItem);
+    }
+
+    private User getAuthenticatedUser() {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        return user;
```
Comment (liliia-ponomarenko) [inline src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java:105]:
```suggestion
    private User getAuthenticatedUser() {
        return (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

File: src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        Set<OrderItem> orderItems = orderItemMapper.toOrderItemSet(shoppingCart.getCartItems());
+        orderItems.forEach(orderItem -> {
+                    if (orderItem.getPrice() == null || orderItem.getQuantity() < 1) {
+                        throw new IllegalArgumentException("OrderItem with id "
+                                + orderItem.getId() + " has a illegal price or quantity.");
+                    }
+                    orderItem.setOrder(order);
+                    orderItem.setPrice(orderItem.getPrice()
+                            .multiply(new BigDecimal(orderItem.getQuantity())));
+                }
```
Comment (liliia-ponomarenko) [inline src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java:58]: better move this logic to the private method ;)

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! A few recommendations ;)

**Pull request merged**